### ISSUE TITLE
Crypt throws errors when using with $this->find('list') or $rules->existsIn()

### ISSUE THREAD

**Issue by shoesole** [UseMuffin/Crypt#4]
When calling $this->find('list') on a model which uses the Crypt behavior I get an error "Call to member function has() on string."

I believe the fix for this is to change line 101 (in CryptBehavior.php) from this:
```
if (($options['fields'] && !in_array($field, (array)$options['fields']))
                    || !$row->has($field)
                )
```

to this: (notice the is_string($field) in the condition
```
if (($options['fields'] && !in_array($field, (array)$options['fields']))
                    || is_string($row) || (is_array($row) ? !isset($row[$field]) : !$row->has($field))
                ) 
```

Let me know if you would prefer I make a pull request to add this change.

**Comment by ADmad** [comment]
Can you provide a test case or example code to reproduce the issue?

**Comment by shoesole** [comment]
[example_code.zip](https://github.com/UseMuffin/Crypt/files/968282/example_code.zip)

Sure thing. I tried to make a test case for this, but unfortunately I was a bit hung up on how to mock the ExistsIn rules checker. I've included some example code to show that will try to save a foreign key to a related model, which will trigger the "ExistsIn" rules checker. This will fail because the $row isn't an object. So $row->has() fails. Let me know if you need more from me. I'll be happy to try and provide it.

**Comment by ADmad** [comment]
Fixed in #6
